Godeps/_workspace/src/github.com/jbenet/go-msgio/msgio.go

```diff
@@ -103,11 +105,16 @@ func (s *writer) WriteMsg(msg []byte) (err error) {
 	defer s.lock.Unlock()
 
 	length := uint32(len(msg))
-	if err := binary.Write(s.W, NBO, &length); err != nil {
+	if err := binary.Write(s.buf, NBO, &length); err != nil {
 		return err
 	}
-	_, err = s.W.Write(msg)
-	return err
+
+	_, err = s.buf.Write(msg)
+	if err != nil {
+		return err
+	}
+
+	return s.buf.Flush()
```

Comment:
ok if the flush happens here that's fine (maybe make a note that the flush must be there or else the write wont write-through to the underlying transport?)